pkg/endpointmanager/manager.go

```diff
@@ -460,6 +464,35 @@ func (mgr *EndpointManager) AddEndpoint(owner regeneration.Owner, ep *endpoint.E
 	return nil
 }
 
+func (mgr *EndpointManager) AddHostEndpoint(ctx context.Context, owner regeneration.Owner,
+	proxy endpoint.EndpointProxy, allocator cache.IdentityAllocator, reason string, nodeName string) (err error) {
+	ep := endpoint.CreateHostEndpoint(owner, proxy, allocator)
+	if err := mgr.AddEndpoint(owner, ep, reason); err != nil {
+		return err
+	}
+
+	epLabels := labels.Labels{}
+	epLabels.MergeLabels(labels.LabelHost)
+
+	if k8s.IsEnabled() {
+		// Retrieve k8s labels.
+		if k8sNode, err := k8s.GetNode(k8s.Client(), nodeName); err != nil {
+			log.WithError(err).Warning("Kubernetes node resource representing own node is not available, cannot set Labels")
+		} else {
+			newLabels := labels.Map2Labels(k8sNode.GetLabels(), labels.LabelSourceK8s)
+			newIdtyLabels, _ := labelsfilter.Filter(newLabels)
+			epLabels.MergeLabels(newIdtyLabels)
+		}
+	}
+
+	// Give the endpoint a security identity
+	newCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
```

Comment:
I'm trying to wrap my head around what this 30 second timeout will do if hit and what will happen. Seems like potentially just a silent ignoring of the label changes..?

Maybe we should check the `newCtx` after we run `ep.UpdateLabels()` to ensure that the context deadline was not exceeded (and log a warning or something, I guess?)